packages/core/src/typegenTypes.ts

```diff
@@ -0,0 +1,260 @@
+import {
+  ActionObject,
+  ActionFunction,
+  Cast,
+  ConditionPredicate,
+  BaseActionObject,
+  DelayConfig,
+  EventObject,
+  InvokeCreator,
+  IndexByType,
+  IsNever,
+  MachineOptions,
+  Values
+} from './types';
+
+export interface TypegenDisabled {
+  '@@xstate/typegen': false;
+}
+export interface TypegenEnabled {
+  '@@xstate/typegen': true;
+}
+export interface TypegenMeta extends TypegenEnabled {
+  matchesStates: string;
+  tags: string;
+  missingImplementations: {
+    actions: string;
+    delays: string;
+    guards: string;
+    services: string;
+  };
+  eventsCausingActions: Record<string, string>;
+  eventsCausingDelays: Record<string, string>;
+  eventsCausingGuards: Record<string, string>;
+  eventsCausingServices: Record<string, string>;
```

Comment:
I would probably prefer combining those under a single property - but didn't come up with good names for this. WDYT?

Reply:
`eventsCausing: { actions: Record<string, string> }`

Is there a benefit to this beyond looking prettier?

Reply:
> Is there a benefit to this beyond looking prettier?

Just this 😅 So ye - probably not worth it. When I think about it now - not nesting them might actually make it easier to index those properties (since I always~ pass the "root" meta-type around).

Reply:
Yep I was thinking that too